### DISCUSSION TITLE
[ZK Stack] - <Cannot restart previously initiated hyperchain>

### DISCUSSION THREAD

**Discussion by nikoumba** [zkSync-Community-Hub/zksync-developers#222] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I've created a hyperchain, closed it and now I cannot restart it.

### Expected Behavior

Expected to have a command like `zk stack reinit`

### Current Behavior

Only available `zk stack init` where it tries to deploy again the contracts

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by githubdoramon** [comment, marked as answer]
Hey @SpiralOutDotEu , thanks for this question. We definitely need to make more clear on the docs, but you can do that easily. I am assuming:
- You used init to configure your hyperchain
- You didn't add the prover
- You didn't create the docker files, you just started in the end of the wizard

Given the above, you can simply run

`zk server --components "http_api,tree_lightweight,eth,data_fetcher,state_keeper,housekeeper"`

And you a good to go.

Let me know how it goes

**Reply by nikoumba** [reply]
Yes it worked, thank you!
I would expect this command to be under `zk stack` since that's where I init the stack

**Comment by jakeaukuso** [comment]
It seems like you're encountering difficulty restarting a closed hyperchain in zkSync. Currently, there isn't a specific command like "zk stack reinit" available for this purpose.

However, you may need to manually manage the state of your hyperchain by ensuring that all necessary steps are taken to properly close and restart it. This might involve reviewing the zkSync documentation or reaching out to the zkSync community for guidance on the appropriate procedures for restarting a closed hyperchain.

**Comment by franksofa** [comment]
To restart a hyperchain after closing it, you can follow these steps:

1. Ensure all components of the hyperchain are properly stopped.
2. Clear any existing data related to the hyperchain.
3. Reinitialize the hyperchain using the appropriate commands or scripts, such as zk stack init.
4. If necessary, configure the hyperchain settings according to your requirements.
5. Start the hyperchain components.

If zk stack init is deploying contracts again, it might be due to the previous initialization not being properly cleared. Ensure all data related to the previous hyperchain initialization is removed before reinitializing.

**Comment by jonnybutlein** [comment]
To restart a hyperchain after it has been closed, you typically need to initialize it again using the appropriate command provided by the zk stack. If there's no specific command like "zk stack reinit," then you may need to follow the initialization process again from scratch, which might involve deploying contracts and setting up configurations anew. Always ensure to backup any necessary data before closing or restarting a hyperchain.

**Comment by antoniostoric** [comment]
Unable to restart a closed hyperchain. Currently, only zk stack init is available, which redeploys contracts.

**Comment by anggabarber** [comment]
It seems like there is an issue with restarting a closed hyperchain on zkSync. The user expects a command like zk stack reinit, but only zk stack init is available, which tries to deploy the contracts again. Without further information or access to the code repository, it is difficult to diagnose the exact issue.
